### DISCUSSION TITLE
Expect a float rather than a different type?

### DISCUSSION THREAD

**Discussion by rm-ccu** [TomSchimansky/CustomTkinter#870] [Q&A] [answered]
*Disclaimer: I'm new(ish) to Python so some of the things I'm trying to do may just not be possible in general.

Below this is just a simple slider but I'm trying to pass floats into it but it doesn't work. (The "print(value)" is just to test)

`def slider(value):
    print(value)

slider = customtkinter.CTkSlider(master=gui, from_=0.1, to=1)
slider.place(relx=0.35, rely=0.05)`

As well, I am trying to take the chosen value in the dialog box and have it determine what part of a function is started but it won't take in any of the values.

`def button_click_eventK():
    dialog = customtkinter.CTkInputDialog(text="Key: ", title="Selector")
    print(dialog.get_input())

kB = customtkinter.CTkButton(gui, text="Text", command=button_click_eventK)
kB.place(relx=0.65, rely=0.45, anchor=tkinter.CENTER)

def start_event():
    if combobox == 'Keyboard':
        print('keyboard')`

**Comment by abhaymakes** [comment]
Are you getting any kind of errors? If yes, then do share.

**Reply by rm-ccu** [reply]
Sorry for the late reply but I had solved it, thanks!

**Comment by Akascape** [comment, marked as answer]
@InkEffect I guess, this is what you need:
```python
def slider(value):
    print(value) # use round(value,2) to round off the float

slider = customtkinter.CTkSlider(master=gui, from_=0.1, to=1, command=slider)
slider.place(relx=0.35, rely=0.05)


def button_click_eventK():
    # pass float from input dialog to slider
    dialog = customtkinter.CTkInputDialog(text="Key: ", title="Selector")
    z = dialog.get_input()
    print(float(z))
    slider.set(float(z))

kB = customtkinter.CTkButton(gui, text="Text", command=button_click_eventK)
kB.place(relx=0.65, rely=0.45, anchor=tkinter.CENTER)
```

**Reply by rm-ccu** [reply]
Not exactly what I wanted but this helped me to realize something I overlooked, thanks a lot!
